packages/babel-parser/src/tokenizer/context.js

```diff
@@ -140,3 +141,8 @@ tt.backQuote.updateContext = function () {
 tt.star.updateContext = function () {
   this.state.exprAllowed = false;
 };
+
+tt.braceHashL.updateContext = function () {
```

Comment:
```suggestion
// we don't need to update context for tt.braceBarL because we do not pop context for tt.braceBarR
tt.braceHashL.updateContext = function () {
```